src/relayclient/RelayClient.ts

```diff
@@ -80,8 +81,34 @@ export class RelayClient {
     configOverride: Partial<GSNConfig>,
     overrideDependencies?: Partial<GSNDependencies>
   ) {
-    this.config = configureGSN(configOverride)
-    const dependencies = getDependencies(this.config, provider, overrideDependencies)
+    this.partialConfig = { provider, configOverride, overrideDependencies }
+    this.logger = console
+  }
+
+  initializing?: Promise<void>
+
+  async init (): Promise<this> {
+    if (this.initializing != null) {
+      await this.initializing
+    }
+    if (this.initialized) {
+      return this
+    }
+    let initializingResolve!: () => void
+    this.initializing = new Promise((resolve) => {
```

Comment:
What's going on here?

Reply:
(renamed "initializing" to "initializingPromise")
1. if currently "initializing", then await for it to finish
2. check if fully initialized, and if so, do nothing
3. create the "initializing" promise, and save its "resolve" in a local variable (initializingResolve)
4. do internalInit
5. call initizlizingResolve to let anyone who entered this method while we're initializing to know its pending